adds core to cloud guide part 2

mirnawong1 opened the pull request:
[ X ] Product approval 
[ X ] Docs approval
[ X ] Must have @azzam34 's final approval before merge


Closes #5252

[notion project](https://www.notion.so/dbtlabs/Move-from-dbt-Core-to-dbt-Cloud-Get-started-c43ddaa1b17c41aab0c66e791ed963b6?pvs=4)

as part of the 3-part-series, this pr adds the 2nd core to cloud guide. 

this guide helps clarify what considerations and technical strategies to consider when planning a move to dbt cloud. 

this pr is merging into current. pr #5158 also needs review and will merge into this one.

[notion doc and project](https://www.notion.so/dbtlabs/Move-from-dbt-Core-to-dbt-Cloud-Everything-you-need-to-know-60dcd8f40b63456ebfea18221ad2e25e)

Commit:
fold in jerco's feedback

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-ff

mirnawong1 committed:
Update core-to-cloud-1.md

adding this as kolv has ga'ed

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

mirnawong1 committed:
Update core-to-cloud-1.md

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-ff

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

mirnawong1 committed:
Update core-to-cloud-1.md

Commit:
add part 2

Commit:
format

mirnawong1 committed:
Update core-to-cloud-1.md

mirnawong1 committed:
Update core-to-cloud-1.md

Commit:
updates

Commit:
updates

Commit:
updates

Commit:
final tweaks

Commit:
updates

Commit:
add faq one

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

mirnawong1 committed:
Update core-to-cloud-1.md

mirnawong1 committed:
Update core-to-cloud-1.md

mirnawong1 committed:
Update core-to-cloud-1.md

Commit:
update to table

mirnawong1 committed:
Update core-to-cloud-1.md

Commit:
updates to guide 2

mirnawong1 committed:
Update core-to-cloud-1.md

Commit:
updates

mirnawong1 committed:
Update core-to-cloud-1.md

Commit:
add links

Commit:
add link

mirnawong1 committed:
Update core-to-cloud-1.md

Commit:
add next

Commit:
jerco's feedback

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

Commit:
update

Commit:
fix link

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into add-core-to-cloud-3

Commit:
fix link

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 committed:
add core to cloud guide 3 (#5233)

as part of the 3-part-series, this pr adds the 3rd core to cloud guide. 

this guide provides users with tips and caveats to help them optimize
their cloud experience.

[notion doc and
project](https://www.notion.so/dbtlabs/Move-from-dbt-Core-to-dbt-Cloud-Optimization-tips-43ee4a509ec143809c605ddeef752031)

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

Commit:
fix

Commit:
fix links

mirnawong1 commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+## Introduction
+Moving from dbt Core to dbt Cloud streamlines analytics engineering workflows by allowing teams to develop, test, deploy, and explore data products using a single, fully managed platform.
+
+Explore our 3-part-guide series on moving from dbt Core to dbt Cloud. The series is ideal for users aiming for streamlined workflows and enhanced analytics:
```
hey @jtcohen6 how does this look to you?  https://docs-getdbt-mnq3cnuh5-dbt-labs.vercel.app/guides/core-to-cloud-2?step=1
<img width="687" alt="Screenshot 2024-04-05 at 13 10 01" src="https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/9a7cf518-2c66-432d-a6cc-1174ca47eb46">

jtcohen6 replied:
I like it! Thank you for incorporating this feedback & running with it :)

In terms of sequence: It feels like "What you need to know" might come _before_ "Get started"?

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+Both development tools are tailored to suit different roles and preferences within your team. To streamline your team’s workflow, it's important to know who is likely to prefer the dbt Cloud IDE and who might lean towards the dbt Cloud CLI. This section is a little different to the others and aims to clarify these preferences.
+
+### dbt Cloud IDE
```
@jtcohen6 i've taken your feedback for the dev tools breakdown and  rather than have a general 'tips and caveats' breakdown, i tried to explain which tools would be best for what user since it's a bit more nuanced and depends on the user.

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/docs/core/installation-overview.md:
```diff
@@ -52,4 +52,7 @@ Most command-line tools, including dbt, have a `--help` flag that you can use to
 :::
 
 ## Related docs
```
I completely understand that we're trying to encourage people to move from Core to Cloud, but putting these migration guides at the bottom of the pages for installing Core and calling them "Related docs" feels a little disingenuous. Maybe we could retitle them something like "Moving from Core to Cloud" or "Upgrading your environment" I'm open to other titles, but calling them "Related docs" right after they get done installing Core feels like we're letting them down. Related docs should lead to more Core workflows.

mirnawong1 replied:
for sure, makes sense to retitle them to 'upgrading your environment'. was trying to  help the guides be more discoverable since we currently have 35 guides.

matthewshaver commented on website/docs/guides/core-to-cloud-1.md:
```diff
@@ -241,7 +241,7 @@ You have learned:
 - How to set up environment variables and validate your models. 
 
-For next steps, we'll soon share other guides on how to manage your move and tips/faqs. Stay tuned!
+Review the next guide, [Move from dbt Core to dbt Cloud: Everything you need to know](/guides/core-to-cloud-2), in this series to learn about the strategic considerations you need to take to move from dbt Core to dbt Cloud.
```
This sounds like something they should read prior to this guide. Putting it at the end is a good idea, but we should possibly put it at the start as well. "If you haven't already read..." or something like that.

mirnawong1 replied:
addressed in this pr: https://github.com/dbt-labs/docs.getdbt.com/pull/5158/files

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+title: 'Move from dbt Core to dbt Cloud: Optimization tips'
+id: core-to-cloud-3
+description: "Use this guide to learn how to optimize your dbt Cloud experience and get answers to common questions."
+hoverSnippet: "se this guide to learn how to optimize your dbt Cloud experience and get answers to common questions."
```
```suggestion
hoverSnippet: "Use this guide to learn how to optimize your dbt Cloud experience and get answers to common questions."
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+<CoretoCloudTable/>
+
+## What you'll learn
+If you're reading this guide, you may have already started your move to dbt Cloud and are looking for tips to help you optimize your dbt Cloud experience, including tips and caveats for the following areas:
```
```suggestion
You may have already started your move to dbt Cloud and are looking for tips to help you optimize your dbt Cloud experience. This guide includes tips and caveats for the following areas:
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+### Caveats
+- Not all parameters are available for adapters.
+- One project can only use one warehouse type.
```
```suggestion
- A project can only use one warehouse type.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+dbt Cloud empowers data practitioners to develop in the tool of their choice. It ships with a [dbt Cloud CLI](/docs/cloud/cloud-cli-installation) (local) or [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud) (browser-based) to build, test, run, and version control your dbt projects.
+
+Both development tools are tailored to suit different roles and preferences within your team. To streamline your team’s workflow, it's important to know who is likely to prefer the dbt Cloud IDE and who might lean towards the dbt Cloud CLI. This section is a little different to the others and aims to clarify these preferences.
```
```suggestion
Both development tools are tailored to suit different audiences and preferences within your team. To streamline your team’s workflow, it's important to know who will prefer the dbt Cloud IDE and who might lean towards the dbt Cloud CLI. This section aims to clarify these preferences.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+Both development tools are tailored to suit different roles and preferences within your team. To streamline your team’s workflow, it's important to know who is likely to prefer the dbt Cloud IDE and who might lean towards the dbt Cloud CLI. This section is a little different to the others and aims to clarify these preferences.
+
+### dbt Cloud IDE
+A single web-based interface for building, testing, running, and version-controlling dbt projects. It compiles dbt code into SQL and executes it directly on your database.
```
```suggestion
A web-based interface for building, testing, running, and version-controlling dbt projects. It compiles dbt code into SQL and executes it directly on your database.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+**Who might prefer the dbt Cloud IDE?**
+
+- New dbt users or those transitioning from other tools who appreciate a more guided experience through a browser-based interface.
+- Team members focused on development speed and convenience, including analysts and less technical stakeholders.
```
A lot of the language here makes the IDE seem like the ideal tool for the new user or junior roles, but some advanced users are going to enjoy using it as well and we should be mindful of that. We might want to call them out in a positive way so they don't feel like using the IDE is somehow making them less than a senior analyst.

"Makes it fast and easy for the most advanced analytics engineers to build and test changes" or something like that.

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+**Key features**
+
+- The dbt Cloud IDE has a simplified Git functionality:
```
```suggestion
- The dbt Cloud IDE has simplified Git functionality:
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+  - View saved but not-committed code changes directly in the IDE.
+- [Format or lint](/docs/cloud/dbt-cloud-ide/lint-format) your code with `sqlfluff` or `sqlfmt`. This includes support for adding your custom linting rules.
+- Allows users to natively [defer to production](/docs/cloud/about-cloud-develop-defer#defer-in-dbt-cloud-cli) metadata directly in their development workflows, reducing the number of objects.
+- Support running multiple dbt commands at the same time through [safe parallel execution](/reference/dbt-commands#parallel-execution), a [feature](/docs/cloud/about-cloud/dbt-cloud-features) available in dbt Cloud's infrastructure. In contrast, `dbt-core` *doesn't support* safe parallel execution for multiple invocations in the same process.
```
```suggestion
- Support for running multiple dbt commands at the same time through [safe parallel execution](/reference/dbt-commands#parallel-execution), a [feature](/docs/cloud/about-cloud/dbt-cloud-features) _only_ available in dbt Cloud. 
```

mirnawong1 commented on website/docs/docs/core/installation-overview.md:
```diff
@@ -52,4 +52,7 @@ Most command-line tools, including dbt, have a `--help` flag that you can use to
 :::
 
 ## Related docs
```
```suggestion
## Upgrading your environment
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- Allows users to natively [defer to production](/docs/cloud/about-cloud-develop-defer#defer-in-dbt-cloud-cli) metadata directly in their development workflows, reducing the number of objects.
+- Support running multiple dbt commands at the same time through [safe parallel execution](/reference/dbt-commands#parallel-execution), a [feature](/docs/cloud/about-cloud/dbt-cloud-features) available in dbt Cloud's infrastructure. In contrast, `dbt-core` *doesn't support* safe parallel execution for multiple invocations in the same process.
+
+The dbt Cloud IDE provides a simplified interface that's accessible to all users, regardless of their technical background. However, there are some capabilities that are intentionally not available in the dbt Cloud IDE due to its focus on simplicity and audience:
```
```suggestion
The dbt Cloud IDE provides a simplified interface that's accessible to all users, regardless of their technical background. However, there are some capabilities that are intentionally not available in the dbt Cloud IDE due to its focus on simplicity and ease of use:
```

mirnawong1 commented on website/docs/docs/core/installation-overview.md:
```diff
@@ -52,4 +52,7 @@ Most command-line tools, including dbt, have a `--help` flag that you can use to
 
 ## Related docs
-- [Move from dbt Core to dbt Cloud](/guides/core-to-cloud-1?step=1)
+- Refer to the 3-part guides series for details, considerations, and tips on how to move to dbt Cloud:
```
```suggestion
Refer to the 3-part guides series for details, considerations, and tips on how to move to dbt Cloud and upgrade your environment:
```

mirnawong1 committed:
Update website/docs/docs/core/installation-overview.md

mirnawong1 committed:
Update website/docs/docs/core/installation-overview.md

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+- Pre-commit for automated checks before *committing* code is not available (yet).
+- Mass-generating files / interacting with the file system are not available.
+- Combining/piping commands, such as `dbt run -s (bash command)` is not available.
```
```suggestion
- Combining/piping commands, such as `dbt run -s (bash command)`, is not available.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- Combining/piping commands, such as `dbt run -s (bash command)` is not available.
+
+### dbt Cloud CLI
+The dbt Cloud CLI allows you to run dbt commands against your dbt Cloud development environment from your local command line. For users who seek full control over their development environment and ideal for those comfortable with the command line.
```
```suggestion
The dbt Cloud CLI allows you to run dbt [commands](/reference/dbt-commands#available-commands) against your dbt Cloud development environment from your local command line. For users who seek full control over their development environment and ideal for those comfortable with the command line.
```

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+**Who might prefer the dbt Cloud IDE?**
+
+- New dbt users or those transitioning from other tools who appreciate a more guided experience through a browser-based interface.
+- Team members focused on development speed and convenience, including analysts and less technical stakeholders.
```
```suggestion
- Team members focused on development speed and convenience.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+**Who might prefer the dbt Cloud CLI?**
+
+- Experienced dbt users and developers looking for granular control over their Git workflows.
```
We're saying that this gives them "full" or "granular" control, so it could be useful to have a section that lists what they can accomplish with the CLI that they can't in the IDE. The key features below seem to be almost the same as the IDE section.

jtcohen6 replied:
This would include all the things that are listed above as not possible with the IDE:
```
- Pre-commit for automated checks before *committing* code is not available (yet).
- Mass-generating files / interacting with the file system are not available.
- Combining/piping commands, such as `dbt run -s (bash command)`, is not available.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+## Orchestration
+
+dbt Cloud provides a robust orchestration that enables users to schedule, run, and monitor dbt jobs with ease. Here are some tips and caveats to consider when using dbt Cloud's orchestration features:
```
```suggestion
dbt Cloud provides robust orchestration that enables you to schedule, run, and monitor dbt jobs with ease. Here are some tips and caveats to consider when using dbt Cloud's orchestration features:
```

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+Both development tools are tailored to suit different audiences and preferences within your team. To streamline your team’s workflow, it's important to know who will prefer the dbt Cloud IDE and who might lean towards the dbt Cloud CLI. This section aims to clarify these preferences.
+
+### dbt Cloud IDE
+A web-based interface for building, testing, running, and version-controlling dbt projects. It compiles dbt code into SQL and executes it directly on your database.
```
```suggestion
A web-based interface for building, testing, running, and version-controlling dbt projects. It compiles dbt code into SQL and executes it directly on your database. The dbt Cloud IDE makes developing fast and easy for new and seasoned data practitioners to build and test changes.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+- Enable [partial parsing](/docs/deploy/deploy-environments#partial-parsing) between jobs in dbt Cloud to significantly speed up project parsing by only processing changed files, optimizing performance for large projects.
+- [Run multiple CI/CD](/docs/deploy/continuous-integration) jobs at the same time which will not block production runs. The Job scheduler automatically cancels stale runs  when a newer commit is pushed. This is because each PR will run in its own schema.
+- dbt Cloud automatically cancels a scheduled run if the existing run is still executing. This prevents unnecessary, duplicative executions.
```
This is more of a notification than a tip. Since it's automatic, does the user need to do any configuration or can we suggest an action for them to make this even more effective?

mirnawong1 replied:
they don't but i thought it'd be helpful to know how it works in cloud. they do need to know that api jobs aren't covered so I'll link to the section.

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+- Enable [partial parsing](/docs/deploy/deploy-environments#partial-parsing) between jobs in dbt Cloud to significantly speed up project parsing by only processing changed files, optimizing performance for large projects.
+- [Run multiple CI/CD](/docs/deploy/continuous-integration) jobs at the same time which will not block production runs. The Job scheduler automatically cancels stale runs  when a newer commit is pushed. This is because each PR will run in its own schema.
+- dbt Cloud automatically cancels a scheduled run if the existing run is still executing. This prevents unnecessary, duplicative executions.
```
```suggestion
- dbt Cloud automatically [cancels](/docs/deploy/job-scheduler#run-cancellation-for-over-scheduled-jobs) a scheduled run if the existing run is still executing. This prevents unnecessary, duplicative executions.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- [Link deploy jobs](/docs/deploy/deploy-jobs#trigger-on-job-completion--) across dbt Cloud projects by configuring your job or using the [Create Job API](/dbt-cloud/api-v2#/operations/Create%20Job) to do this. <Lifecycle status="team,enterprise"/>
+- [Rerun your jobs](/docs/deploy/retry-jobs) from the start or the point of failure if your dbt job run completed with a status of **`Error.`**
+
+### Caveats
```
The items below seem more like they belong in the tips section of the doc. Caveats should serve as a warning of limitations.

mirnawong1 replied:
they're actually more caveats for the scheduler so happy to leave it here!

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+  - Alternatively, check out our [jobs-as-code](https://github.com/dbt-labs/dbt-jobs-as-code) repository, which is a tool built to handle dbt Cloud jobs as a well-defined YAML file.
+- dbt Cloud users and external emails can receive notifications if a job fails, succeeds, or is cancelled. To get notifications for warnings, you can create a [webhook subscription](/guides/zapier-slack) and post to Slack.
+
+## dbt Mesh
```
This guide is labeled as `intermediate`, but isn't mesh more of an advanced workflow? I don't think we need to change the status of the doc, but as this is a "Getting Started" guide, maybe we should call out in the text that this is a more advanced feature?

mirnawong1 replied:
got it, was trying to keep it brief since it links to the docs. i would focus more on mesh being a use case for organizations that are experiencing slow models being built, governance requirements are increasing, teams are working in diff workflows. so the data architecture is helpful for orgs with more mature, complex transformation workflows in dbt. i can add that.

mirnawong1 replied:
ok changed to:

[dbt Mesh](/best-practices/how-we-mesh/mesh-1-intro) helps organizations with mature, complex transformation workflows in dbt increase the flexibility and performance of their dbt projects. It deploys multiple interconnected dbt projects instead of a single large, monolithic project.

It enables you to integrate and navigate between different projects and models with [cross-project dependencies](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref), enhancing collaboration and data governance.

Here are some tips and caveats to consider when using dbt Mesh:

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- Link models across projects for a modular and scalable approach for your project and teams.
+- Manage access to your dbt models both within and across projects using:
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** - Organize nodes in your dbt DAG that share a logical connection and assign an owner to the entire group.
+  - **[Access](/docs/collaborate/govern/model-access#access-modifiers)** - Control who can reference models.
```
```suggestion
  - **[Model Access](/docs/collaborate/govern/model-access#access-modifiers)** &mdash; Control who can reference models.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- To use [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref), all developers need to develop with dbt Cloud (either with the dbt Cloud CLI or dbt Cloud IDE). Cross-project references are not supported in dbt Core.
+- Link models across projects for a modular and scalable approach for your project and teams.
+- Manage access to your dbt models both within and across projects using:
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** - Organize nodes in your dbt DAG that share a logical connection and assign an owner to the entire group.
```
```suggestion
  - **[Groups](/docs/collaborate/govern/model-access#groups)** &mdash; Organize nodes in your dbt DAG that share a logical connection and assign an owner to the entire group.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- Manage access to your dbt models both within and across projects using:
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** - Organize nodes in your dbt DAG that share a logical connection and assign an owner to the entire group.
+  - **[Access](/docs/collaborate/govern/model-access#access-modifiers)** - Control who can reference models.
+  - **[Model Versions](/docs/collaborate/govern/model-versions)** - Enable adoption and deprecation of models as they evolve.
```
```suggestion
  - **[Model Versions](/docs/collaborate/govern/model-versions)** &mdash; Enable adoption and deprecation of models as they evolve.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** - Organize nodes in your dbt DAG that share a logical connection and assign an owner to the entire group.
+  - **[Access](/docs/collaborate/govern/model-access#access-modifiers)** - Control who can reference models.
+  - **[Model Versions](/docs/collaborate/govern/model-versions)** - Enable adoption and deprecation of models as they evolve.
+  - **[Model Contracts](/docs/collaborate/govern/model-contracts)** -Set clear expectations on the shape of the data to ensure data changes upstream of dbt or within a project's logic don't break downstream consumers' data products.
```
```suggestion
  - **[Model Contracts](/docs/collaborate/govern/model-contracts)** &mdash; Set clear expectations on the shape of the data to ensure data changes upstream of dbt or within a project's logic don't break downstream consumers' data products.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+[dbt Explorer](/docs/collaborate/explore-projects) enhances your ability to discover and understand your data models through rich metadata and lineage visualization. Here are some tips and caveats to consider when using dbt Explorer:
+
+### Tips
+- Improve documentation accessibility, making it easier for teams to find and understand the data assets available to them.
```
This is an excellent feature callout, but for this to be a tip we should mention how they would do this (or link to the docs)

mirnawong1 replied:
removed per next comment

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+### Tips
+- Improve documentation accessibility, making it easier for teams to find and understand the data assets available to them.
+- Use the search and filter capabilities in dbt Explorer to quickly locate models, sources, and tests, streamlining your workflow.
```
This is my favorite Explorer feature - maybe we can remove the tip above and make this the first (or combine them)

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- Improve documentation accessibility, making it easier for teams to find and understand the data assets available to them.
+- Use the search and filter capabilities in dbt Explorer to quickly locate models, sources, and tests, streamlining your workflow.
+- View all the [different projects](/docs/collaborate/explore-multiple-projects) and public models in the account, where the public models are defined, and how they are used to gain a better understanding of your cross-project resources.
+- Use the [Lenses](/docs/collaborate/explore-projects#lenses) feature, which are like a map layers for your DAG, available from your project's lineage graph. Lenses help you further understand your project’s contextual metadata at scale, especially to distinguish a particular model or a subset of models.
```
```suggestion
- Use the [Lenses](/docs/collaborate/explore-projects#lenses) feature, which are map-like layers for your DAG, available from your project's lineage graph. Lenses help you further understand your project’s contextual metadata at scale, especially to distinguish a particular model or a subset of models.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+- Access column-level lineage (CLL) for the resources in your dbt project. <Lifecycle status="enterprise"/>
+
+### Caveats
+- There has been at least one successful job run in the production deployment environment.
```
```suggestion
- There must be at least one successful job run in the production deployment environment for Explorer to populate information. 
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+### Caveats
+- There has been at least one successful job run in the production deployment environment.
+- Familiarize yourself with dbt Explorer’s features to fully leverage its capabilities to avoid missed opportunities for efficiency gains.
```
This isn't really a caveat but a suggestion. It's ok to not have multiple caveats if there really is only one. You could potentially make this a closing thought instead.

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+Congratulations on making it through the guide 🎉!
+
+We hope you’re equipped with useful insights and tips to help you with your move. Something to note is that moving from dbt Core to dbt Cloud isn’t just about evolving your data projects — it's about exploring new levels of collaboration, governance, efficiency, and innovation within your team.
```
```suggestion
We hope you’re equipped with useful insights and tips to help you with your move. Something to note is that moving from dbt Core to dbt Cloud isn’t just about evolving your data projects, it's about exploring new levels of collaboration, governance, efficiency, and innovation within your team.
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+We hope you’re equipped with useful insights and tips to help you with your move. Something to note is that moving from dbt Core to dbt Cloud isn’t just about evolving your data projects — it's about exploring new levels of collaboration, governance, efficiency, and innovation within your team.
+
+For the next steps, you can continue exploring our 3-part-guide series on moving from dbt Core to dbt Cloud:
```
```suggestion
For the next steps, continue exploring our 3-part-guide series on moving from dbt Core to dbt Cloud:
```

matthewshaver commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+### Resources
+
+If you need any additional help or have some questions, you can reach out to us in the following ways:
```
```suggestion
If you need any additional help or have some questions, use the following resources:
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+<CoretoCloudTable/>
+
+## What you'll learn
+Migrating from dbt Core to dbt Cloud can yield significant benefits, and a standard way of working. Understanding the differences between dbt Cloud and your current Core deployment will help you strategize and plan for your move.
```
```suggestion
Migrating from dbt Core to dbt Cloud can yield significant benefits and a standardized way of working. Understanding the differences between dbt Cloud and your current Core deployment will help you strategize and plan for your move.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+
+### Caveats
+- There must be at least one successful job run in the production deployment environment for Explorer to populate information. 
+- Familiarize yourself with dbt Explorer’s features to fully leverage its capabilities to avoid missed opportunities for efficiency gains.
```
```suggestion

Familiarize yourself with dbt Explorer’s features to fully leverage its capabilities to avoid missed opportunities for efficiency gains.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,189 @@
+[dbt Explorer](/docs/collaborate/explore-projects) enhances your ability to discover and understand your data models through rich metadata and lineage visualization. Here are some tips and caveats to consider when using dbt Explorer:
+
+### Tips
+- Improve documentation accessibility, making it easier for teams to find and understand the data assets available to them.
```
```suggestion
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+If your team has a custom deployment of dbt Core today, you could be reading this guide because:
+- You’ve realized the burden of maintaining that deployment.
+- The person who set it up has since left.
+- Or you’re interested in what dbt Cloud could do to standardize your deployment and consolidate some of the toolings that you may already have in place.
```
```suggestion
- You’re interested in what dbt Cloud could do to standardize your deployment and consolidate some of the toolings that you may already have in place.
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+- The person who set it up has since left.
+- Or you’re interested in what dbt Cloud could do to standardize your deployment and consolidate some of the toolings that you may already have in place.
+
+This guide shares the technical adjustments and team collaboration strategies you’ll need to know to move your project from dbt Core to dbt Cloud.  Each “BYO” (build your own) deployment of dbt Core will look a little different, but after seeing hundreds of teams make the migration, there are many things in common.
```
BYO has a canonically different meaning and it might be something to avoid
```suggestion
This guide shares the technical adjustments and team collaboration strategies you’ll need to know to move your project from dbt Core to dbt Cloud.  Each "build your own" deployment of dbt Core will look a little different, but after seeing hundreds of teams make the migration, there are many things in common.
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+- How are you scheduling your dbt models to run in production?
+- How are you currently managing Continuous integration/Continuous deployment (CI/CD) of logical changes (if at all)?
+- How do your data developers prefer to work?
+- How do you manage different data environments, and different behaviors in those environments?
```
```suggestion
- How do you manage different data environments and the different behaviors in those environments?
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+- How do you manage different data environments, and different behaviors in those environments?
+
+dbt Cloud provides standard mechanisms for tackling these considerations, all of which delivers long-term benefits to your organization:
+- cross-team collaboration
```
```suggestion
- Cross-team collaboration
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+dbt Cloud provides standard mechanisms for tackling these considerations, all of which delivers long-term benefits to your organization:
+- cross-team collaboration
+- access control
```
```suggestion
- Access control
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+dbt Cloud provides standard mechanisms for tackling these considerations, all of which delivers long-term benefits to your organization:
+- cross-team collaboration
+- access control
+- orchestration
```
```suggestion
- Orchestration
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+- cross-team collaboration
+- access control
+- orchestration
+- isolated data environments
```
```suggestion
- Isolated data environments
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+- orchestration
+- isolated data environments
+
+If you have rolled your own dbt Core deployment, you have probably come up with different answers.
```
```suggestion
If you have rolled out your own dbt Core deployment, you have probably come up with different answers.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+<expandable alt_header="Start small to minimize risk and maximize learning">
+
+You don’t need to move every team and every developer’s workflow all at once. Many customers with large ‘BYO’ (build your own) dbt deployments start by moving one team and one project.
```
```suggestion
You don’t need to move every team and every developer’s workflow all at once. Many customers with large dbt deployments start by moving one team and one project.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+You don’t need to move every team and every developer’s workflow all at once. Many customers with large ‘BYO’ (build your own) dbt deployments start by moving one team and one project.
+
+Once they see the benefits of a consolidated platform, they move the rest of their teams and workflows. While long-term ‘hybrid’ deployments can be challenging, it may make sense as a temporary on-ramp.
```
```suggestion
Once the benefits of a consolidated platform are clear, move the rest of your teams and workflows. While long-term ‘hybrid’ deployments can be challenging, it may make sense as a temporary on-ramp.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+Assess the users or personas involved in the pre-move, during the move, and post-move.
+- **Administrators**: Plan for new access controls in dbt Cloud, such as deciding what teams can manage themselves and what should be standardized. Determine who will be responsible for setting up and maintaining projects, data platform connections, and environments.
+- **Data developers** (Data analysts, Data engineers, Analytics engineers, Business analysts): Determine onboarding order, workflow adaptation in dbt Cloud, training on dbt Cloud CLI or dbt Cloud IDE usage, and role changes.
```
```suggestion
- **Data developers** (data analysts, data engineers, analytics engineers, business analysts): Determine onboarding order, workflow adaptation in dbt Cloud, training on dbt Cloud CLI or dbt Cloud IDE usage, and role changes.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+<expandable alt_header="Onboarding order"> 
+
+If you have multiple teams of dbt developers, think about how to start your onboarding sequence for dbt Cloud.
+- Start with downstream (like business-embedded teams) who may benefit from the dbt Cloud IDE as dev experience (less technical users) and sharing features (like Auto-deferral and dbt Explorer) to share with their stakeholders, moving to more technical teams later.
```
```suggestion
- Start with downstream (like business-embedded teams) who may benefit from the dbt Cloud IDE as dev experience (less technical users) and sharing features (like auto-deferral and dbt Explorer) to share with their stakeholders, moving to more technical teams later.
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+<expandable alt_header="Onboarding order"> 
+
+If you have multiple teams of dbt developers, think about how to start your onboarding sequence for dbt Cloud.
```
```suggestion
If you have multiple teams of dbt developers, think about how to start your onboarding sequence for dbt Cloud:
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+<expandable alt_header="Analyze current workflows, review processes, and team structures">
+
+Discover how dbt Cloud can help simplify development, orchestration, and testing.
```
```suggestion
Discover how dbt Cloud can help simplify development, orchestration, and testing:
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+- **Orchestration**: Create custom schedules to run your production jobs. Schedule jobs by day of the week, time of day, or a recurring interval.
+  - Set up CI to ensure developer effectiveness, and CD jobs to deploy changes as soon as they’re merged. 
+  - Link deploy jobs together by [triggering a job](/docs/deploy/deploy-jobs#trigger-on-job-completion) when another one is completed. 
+  - For the most flexibility, you can use the [dbt Cloud API](https://docs.getdbt.com/dbt-cloud/api-v2#/) to trigger jobs. This makes sense when you want to integrate dbt execution with other data workflows.
```
```suggestion
  - For the most flexibility, use the [dbt Cloud API](https://docs.getdbt.com/dbt-cloud/api-v2#/) to trigger jobs. This makes sense when you want to integrate dbt execution with other data workflows.
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+Transition to dbt Cloud's [access control](/docs/cloud/manage-access/about-user-access) mechanisms to ensure security and proper access management. dbt Cloud administrators can use dbt Cloud's permission model to control user-level access in a dbt Cloud account:
+- **License-based access controls:** Users are configured with account-wide license types. These licenses control the things a user can do within the application: view project metadata, develop changes within those projects, or administer access to those projects.
+- **Role-based Access Control (RBAC):** Users are assigned to *groups* that have specific permissions on specific projects or all projects in the account. A user may be a member of multiple groups, and those groups may have permissions on multiple projects.
```
```suggestion
- **Role-based Access Control (RBAC):** Users are assigned to *groups* with specific permissions on specific projects or all projects in the account. A user may be a member of multiple groups, and those groups may have permissions on multiple projects. <Lifecycle status="enterprise" />
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+<expandable alt_header="Manage environments"> 
+
+If you require isolation between production and non-production data environments due to sensitive data, dbt Cloud can support Development, Staging (_soon_), and Production data environments.
```
```suggestion
If you require isolation between production and pre-production data environments due to sensitive data, dbt Cloud can support Development, Staging (_soon_), and Production data environments.
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+If you require isolation between production and non-production data environments due to sensitive data, dbt Cloud can support Development, Staging (_soon_), and Production data environments.
+
+This provides developers with the benefits of an enhanced workflow while ensuring isolation between Staging and Production data, and locking down permissions on Prod.
```
I don't think we really treat production and staging as proper nouns elsewhere in the docs. Is there a particular reason for doing so here?

mirnawong1 replied:
i _think_ this is related to what's coming soon in terms of the diff environments. going to loop @jtcohen6 as i thought it was related which is why i kept it capitalized.

jtcohen6 replied:
@matthewshaver This is related to the new "Staging" type of deployment environment that we're rolling out alongside "Production! I think you're driving that one ([internal notion](https://www.notion.so/dbtlabs/dbt-Cloud-staging-environment-684abeb5baf24b6fbd55bd5c9f56b606?pvs=31)). Agree that proper nouns make more sense here.

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+## Move to dbt Cloud
+
+After reviewing the considerations and planning your move, you may want to start moving your dbt Core project to dbt Cloud &mdash; check out the detailed [Move to dbt Cloud: Get started](/guides/core-to-cloud-1?step=1) guide for useful tasks and insights for a smooth move from dbt Core to dbt Cloud.
```
```suggestion
After reviewing the considerations and planning your move, you may want to start moving your dbt Core project to dbt Cloud. Check out the detailed [Move to dbt Cloud: Get started](/guides/core-to-cloud-1?step=1) guide for useful tasks and insights for a smooth transition from dbt Core to dbt Cloud.
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+After reviewing the considerations and planning your move, you may want to start moving your dbt Core project to dbt Cloud &mdash; check out the detailed [Move to dbt Cloud: Get started](/guides/core-to-cloud-1?step=1) guide for useful tasks and insights for a smooth move from dbt Core to dbt Cloud.
+
+This guide is your roadmap to help you think about migration strategies and what moving from dbt Core to dbt Cloud would look like. 
```
```suggestion
This guide is your roadmap to help you think about migration strategies and what moving from dbt Core to dbt Cloud could look like. 
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+- **Review your dbt project:** Ensure your project compiles correctly and that you can run commands. Make sure your models are accurate and monitor performance post-move.
+- **Start cutover:** You can start the cutover to dbt Cloud by creating a dbt Cloud job with commands that only run a small subset of the DAG. Validate the tables are being populated in the proper database/schemas as expected. Then continue to expand the scope of the job to include more sections of the DAG as you gain confidence in the results.
+- **Precision in testing:** Use [unit testing](/docs/build/unit-tests) to allow you to validate your SQL modeling logic on a small set of static inputs *before* you materialize your full model in production.
```
```suggestion
- **Precision testing:** Use [unit testing](/docs/build/unit-tests) to allow you to validate your SQL modeling logic on a small set of static inputs *before* you materialize your full model in production.
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+Once you’ve confirmed that dbt Cloud orchestration and CI/CD are working as expected, you should pause your current orchestration tool and stop or update your current CI/CD process. This is not relevant if you’re still using an external orchestrator (such as Airflow), and you’ve swapped out `dbt-core` execution for dbt Cloud execution (through the [API](/docs/dbt-cloud-apis/overview)).
+
+Familiarize your team with dbt Cloud's [features](/docs/cloud/about-cloud/dbt-cloud-features) and optimize development and deployment processes. Some key features to consider include:
+- **Version management:** Manage [dbt versions](/docs/dbt-versions/upgrade-dbt-version-in-cloud) and ensure team synchronization with dbt Cloud's one-click feature, eliminating the hassle of manual updates and version discrepancies. You can also **[Keep on latest version](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version)** to always get the latest fixes and early access to n ...
```
Does "Keep on latest version" technically give them early access? I thought that was the experimental feature toggle. 
```suggestion
- **Version management:** Manage [dbt versions](/docs/dbt-versions/upgrade-dbt-version-in-cloud) and ensure team synchronization with dbt Cloud's one-click feature, eliminating the hassle of manual updates and version discrepancies. You can also **[Keep on latest version](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version)** to always get the latest fixes and access to new functionality for your dbt project as soon as it launches.
```

mirnawong1 replied:
i saw this on the page here: https://docs.getdbt.com/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version so will cc @jtcohen6 to confirm!

jtcohen6 replied:
Yes, earlier access is correct. Cloud customers on KoLV can try out / leverage "core" features before they land in a final release of `dbt-core` OSS.

mirnawong1 replied:
resolving this

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+Familiarize your team with dbt Cloud's [features](/docs/cloud/about-cloud/dbt-cloud-features) and optimize development and deployment processes. Some key features to consider include:
+- **Version management:** Manage [dbt versions](/docs/dbt-versions/upgrade-dbt-version-in-cloud) and ensure team synchronization with dbt Cloud's one-click feature, eliminating the hassle of manual updates and version discrepancies. You can also **[Keep on latest version](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version)** to always get the latest fixes and early access to n ...
+- **Development tools**: Use the [dbt Cloud CLI](/docs/cloud/cloud-cli-installation) or [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud) to build, test, run, and version control your dbt projects.
+- **Documentation and Source freshness:**  Automate storage of documentation and track [source freshness](/docs/deploy/source-freshness) in dbt Cloud, streamlining project maintenance.
```
```suggestion
- **Documentation and Source freshness:**  Automate storage of [documentation](/docs/collaborate/documentation) and track [source freshness](/docs/deploy/source-freshness) in dbt Cloud, streamlining project maintenance.
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+Now that you’ve chosen dbt Cloud as your platform, you’ve unlocked the power of streamlining collaboration, enhancing workflow efficiency, and leveraging powerful [features](/docs/cloud/about-cloud/dbt-cloud-features) for analytics engineering teams. Here are some additional features you can use to unlock the full potential of dbt Cloud:
+
+- **Audit logs:** Use [audit logs](/docs/cloud/manage-access/audit-log) to review actions performed by people in your organization. Audit logs contain audited user and system events in real time. You can even export *all* the activity (beyond the 90 days you can view in dbt Cloud). <Lifecycle status="enterprise"/>
```
```suggestion
- **Audit logs:** Use [audit logs](/docs/cloud/manage-access/audit-log) to review actions performed by people in your organization. Audit logs contain audited user and system events in real time. You can even [export](/docs/cloud/manage-access/audit-log#exporting-logs) *all* the activity (beyond the 90 days you can view in dbt Cloud). <Lifecycle status="enterprise"/>
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+Now that you’ve chosen dbt Cloud as your platform, you’ve unlocked the power of streamlining collaboration, enhancing workflow efficiency, and leveraging powerful [features](/docs/cloud/about-cloud/dbt-cloud-features) for analytics engineering teams. Here are some additional features you can use to unlock the full potential of dbt Cloud:
+
+- **Audit logs:** Use [audit logs](/docs/cloud/manage-access/audit-log) to review actions performed by people in your organization. Audit logs contain audited user and system events in real time. You can even export *all* the activity (beyond the 90 days you can view in dbt Cloud). <Lifecycle status="enterprise"/>
+- **dbt Cloud APIs:** Use dbt Cloud's robust [APIs](/docs/dbt-cloud-apis/overview) to Create, Read, Update, and Delete (CRUD) projects/jobs/environments project. The [dbt Cloud Administrative API](/docs/dbt-cloud-apis/admin-cloud-api) and [Terraform provider](https://registry.terraform.io/providers/dbt-labs/dbtcloud/latest/docs/resources/job) facilitate programmatic access and configuration stora ...
```
```suggestion
- **dbt Cloud APIs:** Use dbt Cloud's robust [APIs](/docs/dbt-cloud-apis/overview) to create, read, update, and delete (CRUD) projects/jobs/environments project. The [dbt Cloud Administrative API](/docs/dbt-cloud-apis/admin-cloud-api) and [Terraform provider](https://registry.terraform.io/providers/dbt-labs/dbtcloud/latest/docs/resources/job) facilitate programmatic access and configuration storage. While the [Discovery API](/docs/dbt-cloud-apis/discovery-api) offers extensive metadata querying capabilities, such as job data, model configurations, usage, and overall project health. <Lifecycle status="team,enterprise"/>
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+- **Audit logs:** Use [audit logs](/docs/cloud/manage-access/audit-log) to review actions performed by people in your organization. Audit logs contain audited user and system events in real time. You can even export *all* the activity (beyond the 90 days you can view in dbt Cloud). <Lifecycle status="enterprise"/>
+- **dbt Cloud APIs:** Use dbt Cloud's robust [APIs](/docs/dbt-cloud-apis/overview) to Create, Read, Update, and Delete (CRUD) projects/jobs/environments project. The [dbt Cloud Administrative API](/docs/dbt-cloud-apis/admin-cloud-api) and [Terraform provider](https://registry.terraform.io/providers/dbt-labs/dbtcloud/latest/docs/resources/job) facilitate programmatic access and configuration stora ...
+- **dbt Explorer**: Use [dbt Explorer](/docs/collaborate/explore-projects) to view your project's [resources](/docs/build/projects) (such as models, tests, and metrics) and their [lineage](https://docs.getdbt.com/terms/data-lineage) to gain a better understanding of its latest production state. (Once you have a successful job in a Production environment). <Lifecycle status="team,enterprise"/>
+- **dbt Semantic Layer:** The [dbt Semantic Layer](/docs/use-dbt-semantic-layer/dbt-sl) allows you to define universal metrics on top of your models, that can then be queried in your business intelligent (BI) tool. This means no more inconsistent metrics — there’s now a centralized way to define these metrics and create visibility in every component of the data flow. <Lifecycle status="team,enter ...
```
```suggestion
- **dbt Semantic Layer:** The [dbt Semantic Layer](/docs/use-dbt-semantic-layer/dbt-sl) allows you to define universal metrics on top of your models that can then be queried in your business intelligence (BI) tool. This means no more inconsistent metrics — there’s now a centralized way to define these metrics and create visibility in every component of the data flow. <Lifecycle status="team,enterprise"/>
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+## Summary
+
+This guide should now have given you some insight and equipped you with a framework for moving from dbt Core to dbt Cloud. This guide has covered the following key areas:
```
```suggestion
You have now been given insight and equipped with a framework for moving from dbt Core to dbt Cloud. This guide has covered the following key areas:
```

matthewshaver commented on website/docs/guides/core-to-cloud-2.md:
```diff
@@ -0,0 +1,196 @@
+
+<CoretoCloudTable/>
+
+### Related docs
```
Since it's not just docs
```suggestion
### Related content
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-2.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

jtcohen6 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,188 @@
+**Who might prefer the dbt Cloud IDE?**
+
+- New dbt users or those transitioning from other tools who appreciate a more guided experience through a browser-based interface.
+- Team members focused on development speed and convenience.
```
Maybe:
```suggestions
- Team members focused on speed and convenience for getting started with a new or existing project.
```

mirnawong1 committed:
Update _core-to-cloud-guide-table.md

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

Commit:
matt/jerco's feedback

Commit:
Merge branch 'core-to-cloud-2' of https://github.com/dbt-labs/docs.getdbt.com into core-to-cloud-2

mirnawong1 committed:
Update about-core-setup.md

mirnawong1 committed:
Update installation-overview.md

removing per matt's feedback

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-1.md

Commit:
update order

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

Commit:
remove position

Commit:
add id

Commit:
fix width

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,192 @@
+- Use [dbt-meshify](https://github.com/dbt-labs/dbt-meshify) to separate your mono-repo and move to dbt Cloud.
+
+### Caveat
+- Currently, dbt Mesh does not support circular project dependencies.
```
@jtcohen6 i think you mentioned you had some more bits to add here for mesh?

jtcohen6 replied:
@mirnawong1 Here are the top three I would list. Probably too much text, so feel free to cut it down.

### Caveats

**Use of cross-project references requires that each upstream dbt project maps to exactly one dbt Cloud project.** We strongly discourage defining multiple projects for the same codebase, whether for the purposes of managing permissions, data warehouse connections, and isolation of production from non-production data. While this was required historically, this should no longer be necessary with the rollout of Staging environments (beta, link), Environment-level RBAC (coming soon), and Extended Attributes (link).

**Project dependencies are uni-directional.** dbt will check for cycles across projects and raise errors if any are detected. We are considering support for bi-directional project dependencies in the future, whereby dbt would still check for node-level cycles while allowing cycles at the project level.

**Public model metadata is visible to everyone in the account.** The goal is to improve discoverability of data products. This is different from access to the underlying data, which is managed by permissions in the data warehouse. For use cases where even metadata about a reusable data asset is sensitive, we are considering an optional extension of `protected` models (https://github.com/dbt-labs/dbt-core/issues/9340).

For more questions, see the FAQs on dbt Mesh: https://docs.getdbt.com/best-practices/how-we-mesh/mesh-4-faqs

Commit:
clarify

Commit:
add

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,192 @@
+You may have already started your move to dbt Cloud and are looking for tips to help you optimize your dbt Cloud experience. This guide includes tips and caveats for the following areas:
+
+- [Adapters and connections](https://docs.getdbt.com/guides/core-to-cloud-3?step=3) 
+- [Development tools](https://docs.getdbt.com)/guides/core-to-cloud-3?step=4) 
```
```suggestion
- [Development tools](https://docs.getdbt.com/guides/core-to-cloud-3?step=4) 
```

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

mirnawong1 committed:
Merge branch 'core-to-cloud-2' into core-to-cloud-ff

mirnawong1 committed:
core to cloud fast follows (#5158)

this pr folds in @jtcohen6's feedback for the first core to cloud guide.
https://www.notion.so/dbtlabs/For-Jeremy-s-review-d0d92d1b8c08496ea19800063cbed8bb

this pr merges into #5232

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

Commit:
nits

jtcohen6 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,192 @@
+- Link models across projects for a modular and scalable approach for your project and teams.
+- Manage access to your dbt models both within and across projects using:
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** &mdash; Organize nodes in your dbt DAG that share a logical connection and assign an owner to the entire group.
+  - **[Model Access](/docs/collaborate/govern/model-access#access-modifiers)** &mdash; Control who can reference models.
```
```suggestion
  - **[Model access](/docs/collaborate/govern/model-access#access-modifiers)** &mdash; Control which other models or projects can reference this model.
```

jtcohen6 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,192 @@
+Here are some tips and caveats to consider when using dbt Mesh:
+
+### Tips
+- To use [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref), all developers need to develop with dbt Cloud (either with the dbt Cloud CLI or dbt Cloud IDE). Cross-project references are not supported in dbt Core.
```
```suggestion
- To dynamically resolve [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref), all developers need to develop with dbt Cloud (either with the dbt Cloud CLI or dbt Cloud IDE). Cross-project references aren't natively supported in dbt Core, except by installing the source code from upstream projects [as packages](/docs/build/packages#how-do-i-add-a-package-to-my-project)
```

jtcohen6 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,192 @@
+  - **[Model Access](/docs/collaborate/govern/model-access#access-modifiers)** &mdash; Control who can reference models.
+  - **[Model Versions](/docs/collaborate/govern/model-versions)** &mdash; Enable adoption and deprecation of models as they evolve.
+  - **[Model Contracts](/docs/collaborate/govern/model-contracts)** &mdash; Set clear expectations on the shape of the data to ensure data changes upstream of dbt or within a project's logic don't break downstream consumers' data products.
+- Use [dbt-meshify](https://github.com/dbt-labs/dbt-meshify) to separate your mono-repo and move to dbt Cloud.
```
```suggestion
- Use [dbt-meshify](https://github.com/dbt-labs/dbt-meshify) to accelerate splitting apart your monolith into multiple projects.
```

jtcohen6 reviewed (commented):
@mirnawong1 Left a more thorough review of the Mesh section. Thanks again for pulling this together!

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Jeremy Cohen <jeremy@dbtlabs.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Jeremy Cohen <jeremy@dbtlabs.com>

mirnawong1 committed:
Update website/docs/guides/core-to-cloud-3.md

Co-authored-by: Jeremy Cohen <jeremy@dbtlabs.com>

mirnawong1 commented:
brilliant, thank you @jtcohen6 - folded in your suggestions and added the caveats as so:

### Caveats
- To use cross-project references in dbt, each dbt project must correspond to just one dbt Cloud project. We strongly discourage defining multiple projects for the same codebase, even if you're trying to manage access permissions, connect to different data warehouses, or separate production and non-production data.  While this was required historically, upcoming features like Staging environments (_coming soon_), Environment-level RBAC (_coming soon_), and [Extended attributes](/docs/dbt-cloud-environments#extended-attributes) will make it unnecessary.
- Project dependencies are uni-directional, meaning they go in one direction. This means dbt checks for cycles across projects (circular dependencies) and raise errors if any are detected. However, we are considering support to allow projects to depend on each other in both directions in the future, with dbt still checking for node-level cycles while allowing cycles at the project level.
- Everyone in the account can view public model metadata, which helps users find data products more easily. This is separate from who can access the actual data, which is controlled by permissions in the data warehouse. For use cases where even metadata about a reusable data asset is sensitive, we are [considering](dbt-labs/dbt-core#9340) an optional extension of protected models.

Refer to the [dbt Mesh FAQs](/best-practices/how-we-mesh/mesh-4-faqs) for more questions.

Commit:
jerco's feedback

Commit:
fix link

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,196 @@
+dbt Cloud empowers data practitioners to develop in the tool of their choice. It ships with a [dbt Cloud CLI](/docs/cloud/cloud-cli-installation) (local) or [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud) (browser-based) to build, test, run, and version control your dbt projects.
+
+Both development tools are tailored to suit different audiences and preferences within your team. To streamline your team’s workflow, it's important to know who will prefer the dbt Cloud IDE and who might lean towards the dbt Cloud CLI. This section aims to clarify these preferences.
+
```
@jtcohen6 quick one - based on [this thread](https://dbt-labs.slack.com/archives/C012TTNN0JK/p1712760453549829), should we also add info about the gitignore file as described [here](https://docs.getdbt.com/docs/collaborate/git/version-control-basics#the-gitignore-file)

> Make sure you check the .gitignore file contains the necessary entries. dbt Core doesn't interact with git so dbt Cloud doesn't automatically add or verify entries in the .gitignore file. Additionally, if the repository already contains dbt code and doesn't require initialization, dbt Cloud won't add any missing entries to the .gitignore file.

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

mirnawong1 commented on website/docs/guides/core-to-cloud-3.md:
```diff
@@ -0,0 +1,196 @@
+### dbt Cloud CLI
+The dbt Cloud CLI allows you to run dbt [commands](/reference/dbt-commands#available-commands) against your dbt Cloud development environment from your local command line. For users who seek full control over their development environment and ideal for those comfortable with the command line.
+
+**Who might prefer the dbt Cloud CLI?**
```
added this here @jtcohen6

mirnawong1 replied:
When moving from dbt Core to dbt Cloud, make sure you check the `.gitignore` file contains the [necessary folders](/docs/collaborate/git/version-control-basics#the-gitignore-file). dbt Core doesn't interact with git so dbt Cloud doesn't automatically add or verify entries in the `.gitignore` file. Additionally, if the repository already contains dbt code and doesn't require initialization, dbt Cloud won't add any missing entries to the `.gitignore file`.

Commit:
add'l info

mirnawong1 committed:
Update manual-install-qs.md

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

Commit:
azzams feedback

Commit:
add pmm feedback

mirnawong1 commented on website/docs/guides/core-cloud-2.md:
```diff
@@ -0,0 +1,197 @@
+- [Considerations](https://docs.getdbt.com/guides/core-cloud-2?step=3): Learn about the most important things you need to think about when moving from Core to Cloud.
+- [Plan your move](https://docs.getdbt.com/guides/core-cloud-2?step=4): Considerations you need to make, such as user roles and permissions, onboarding order, current workflows, and more.
+- [Move to dbt Cloud](https://docs.getdbt.com/guides/core-cloud-2?step=5): Review the steps to move your dbt Core project to dbt Cloud, including setting up your account, data platform, and Git repository.
+- [Test and validate](https://docs.getdbt.com/guides/core-cloud-2?step=6): Discover how to ensure model accuracy and performance post-move,.
```
```suggestion
- [Test and validate](https://docs.getdbt.com/guides/core-cloud-2?step=6): Discover how to ensure model accuracy and performance post-move.
```

mirnawong1 committed:
Update core-cloud-2.md

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2

Commit:
final tweaks

Commit:
final tweaks

Commit:
fix link

mirnawong1 committed:
Merge branch 'current' into core-to-cloud-2
